plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/VifDriverBase.java

```diff
@@ -44,15 +44,15 @@ public void configure(Map<String, Object> params) throws ConfigurationException
         _pifs = (Map<String, String>)params.get("libvirt.host.pifs");
     }
 
-    protected String getControllCidr(String defaultValue, Map<String, Object> params) {
-        String controllCidr = AgentPropertiesFileHandler.getPropertyValue(AgentProperties.CONTROLL_CIDR);
-        if (StringUtils.isBlank(controllCidr)) {
-            controllCidr = (String) params.get("control.cidr");
+    protected String getControlCidr(String defaultValue, Map<String, Object> params) {
+        String controlCidr = AgentPropertiesFileHandler.getPropertyValue(AgentProperties.CONTROL_CIDR);
+        if (StringUtils.isBlank(controlCidr)) {
+            controlCidr = (String) params.get("control.cidr");
```

Comment:
I think this is not needed. The AgentPropertiesFileHandler class should provide us what's in the agent.properties file or return the default.

Do we maybe want to add some debug logging here as well?

Reply:
I think the value may also come from the MS, so if this is not configured (for a certain agent) the global configuration is used.

Reply:
params.get() does not fetch anything from the MS, right?

Besides that, the cloud0 bridge is configured before the MS is contacted, so any value from from the MS is useless at that point.

Reply:
I'll go through the flow once more, but as I recall params is not only filled from local (agent.properties/env) configuration but also from the values supplied in the commands from the MS. So `params.get()` does not retrieve from the server but `params` already contains values sent by the server.

Reply:
You might be right, however, once the cloud0 bridge is configured by the agent it's never touched again. This happens before it connects to the mgmt server.